lib/tether/interfaces.go

```diff
@@ -44,6 +48,12 @@ type Operations interface {
 	// Returns a function to invoke after the session state has been persisted
 	HandleSessionExit(config *ExecutorConfig, session *SessionConfig) func()
 	ProcessEnv(env []string) []string
+	// LaunchUtility starts a process and provides a way to block on completion and retrieve
+	// it's exit code. This is needed to co-exist with a childreaper.
+	LaunchUtility(UtilityFn) (<-chan int, error)
```

Comment:
I think it would make sense to start using utilities which generate interfaces from the existing data structure: https://github.com/vburenin/ifacemaker - this is a good one :-)